src/payment/strategies/bluesnapv2/bluesnapv2-payment-page-loader.ts

```diff
@@ -0,0 +1,59 @@
+import { createFormPoster } from '@bigcommerce/form-poster';
+
+import { NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { PaymentMethodCancelledError } from '../../errors';
+
+import { BlueSnapV2PaymentInitializeOptions } from './bluesnapv2-payment-options';
+
+export default class BlueSnapV2PaymentPageLoader {
+    private _blueSnapV2Options?: BlueSnapV2PaymentInitializeOptions;
+    private _formPoster = createFormPoster();
+
+    initialize(options?: BlueSnapV2PaymentInitializeOptions): void {
+        this._blueSnapV2Options = options;
+    }
+
+    loadPaymentPage(frameName: string): Promise<undefined> {
+        return new Promise((_, reject) => {
+            if (!this._blueSnapV2Options) {
+                throw new NotInitializedError(
+                    NotInitializedErrorType.PaymentNotInitialized
+                );
+            }
+
+            const { addFrame } = this._blueSnapV2Options;
+            addFrame(undefined, this._createFrame(frameName), () => {
+                reject(new PaymentMethodCancelledError());
+            });
+
+            window.addEventListener('message', this._receiveMessage, false);
+        });
+    }
+
+    private _receiveMessage = ({ data: { action, form } }: MessageEvent): void => {
+        if (action === 'setExternalCheckout') {
+            window.removeEventListener('message', this._receiveMessage, false);
+
+            if (!this._blueSnapV2Options) {
+                throw new NotInitializedError(
+                    NotInitializedErrorType.PaymentNotInitialized
+                );
+            }
+            const { removeFrame } = this._blueSnapV2Options;
+            removeFrame();
```

Comment:
the action that is happening here seems to be the form being submitted. Should this be called `onSubmit` instead of `removeFrame` ? it's up to the client to decide what should be done when the form is submitted.

Reply:
I'm afraid I didn't understand this part 🤔

Reply:
Anyway, I removed that part. Thanks.